qlib/kernel/asm/aarch64.rs

```diff
@@ -524,3 +535,30 @@ pub unsafe fn midr() -> u64 {
     asm!("mrs {0}, midr_el1", out(reg) ret);
     ret
 }
+
+
+pub unsafe fn sp_el0() -> u64 {
+    let ret: u64;
+    asm!("mrs {0}, sp_el0", out(reg) ret);
+    ret
+}
+
+pub unsafe fn sp_el0_write(val: u64) {
+    asm!("msr sp_el0, {0}", in(reg) val);
+}
+
+pub unsafe fn sp_el1() -> u64 {
```

Comment:
since reading/writing SP_EL1 via MRS/MSR is illegal in EL1, and qkernel does not go beyond EL1, perhaps we can remove the function `sp_el1` and `ap_el1_write` since they won't work?

Reply:
removed